sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsSubscriptionKeyCredential.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics.models;
+
+/**
+ * Subscription key credential that shared across cognitive services, or restrict to single service.
+ */
+public final class TextAnalyticsSubscriptionKeyCredential {
+    private String subscriptionKey;
+
+    /**
+     * Creates a {@code SharedKeyCredential} model that describes subscription key.
+     *
+     * @param subscriptionKey the subscription key
+     */
+    public TextAnalyticsSubscriptionKeyCredential(String subscriptionKey) {
+        this.subscriptionKey = subscriptionKey;
+    }
+
+    /**
+     * Get the subscription key value.
+     *
+     * @return the subscription key value
+     */
+    public String getSubscriptionKey() {
```

Comment:
Is there a reason why we need this? 
Although the guidelines are not done, during the meeting they agreed that 
>SHOULD NOT define public properties or fields which allow users to access the authentication data directly.

Reply:
Yes. the property or field "private String subscriptionKey" is a private. User have to invoke the setter `updateCredential` to update the authentication data.

Reply:
thanks for clarifying! this is me not knowing Java :P